skyfield/positionlib.py

```diff
@@ -152,18 +151,61 @@ def separation_from(self, another_icrf):
         c = dots(u1, u2)
         return Angle(radians=arccos(clip(c, -1.0, 1.0)))
 
-    def ecliptic_position(self):
-        """Compute J2000 ecliptic coordinates (x, y, z)"""
-        vector = _ECLIPJ2000.dot(self.position.au)
-        return Distance(vector)
+    def ecliptic_xyz(self, epoch=None):
+        """Compute J2000 ecliptic coordinates (x, y, z)
 
-    def ecliptic_latlon(self):
-        """Compute J2000 ecliptic coordinates (lat, lon, distance)"""
-        vector = _ECLIPJ2000.dot(self.position.au)
-        d, lat, lon = to_polar(vector)
-        return (Angle(radians=lat, signed=True),
-                Angle(radians=lon),
-                Distance(au=d))
+        If you instead want the coordinates referenced to the dynamical
+        system defined by the Earth's true equator and equinox, provide
+        an epoch time.
+        """
+        lat, lon, d = self.ecliptic_latlon(epoch)
+        vector = from_polar(d.au, lat.radians, lon.radians)
+        if len(vector.shape) is 1:
+            return Distance(vector)
+        else:
+            return Distance(vector.T)
+
+    def ecliptic_latlon(self, epoch=None):
+        """Compute J2000 ecliptic coordinates (lat, lon, distance)
+
+        If you instead want the coordinates referenced to the dynamical
+        system defined by the Earth's true equator and equinox, provide
+        an epoch time.
+        """
+        position_au = self.position.au
+        if epoch is not None:
+            if isinstance(epoch, Time):
+                pass
+            elif isinstance(epoch, float):
+                epoch = Time(None, tt=epoch)
+            elif epoch == 'date':
+                epoch = self.t
+            else:
+                raise ValueError('the epoch= must be a Time object,'
+                                 ' a floating point Terrestrial Time (TT),'
+                                 ' or the string "date" for epoch-of-date')
+            position_au = einsum('ij...,j...->i...', epoch.M, position_au)
+            if not epoch.shape:
+                vector = epoch.E.dot(position_au)
+                d, lat, lon = to_polar(vector)
+                return (Angle(radians=lat, signed=True),
```

Comment:
Since every `if` ends with this same routine to build the `Angle` the same way, it feels like instead of repeating the code 3 times that you could put it down at the end of the routine outside of the `if`'s. Does that look possible to you as well?

Reply:
Ok.